indexing-hadoop/src/main/java/io/druid/indexer/updater/HadoopConverterJob.java

```diff
@@ -0,0 +1,1166 @@
+    protected static final String SHOULD_COMMIT_METADATA_KEY = "io.druid.indexer.updater.converter.shouldCommitMetadata";
+    protected static final String PUBLISHED_SEGMENT_KEY = "io.druid.indexer.updater.converter.publishedSegment";
+    private static final Logger log = new Logger(ConvertingOutputFormat.class);
+
+    @Override
+    public RecordWriter<Text, Text> getRecordWriter(TaskAttemptContext context) throws IOException, InterruptedException
+    {
+      return new RecordWriter<Text, Text>()
+      {
+        @Override
+        public void write(Text key, Text value) throws IOException, InterruptedException
+        {
+          // NOOP
+        }
+
+        @Override
+        public void close(TaskAttemptContext context) throws IOException, InterruptedException
+        {
+          // NOOP
+        }
+      };
+    }
+
+    @Override
+    public void checkOutputSpecs(JobContext context) throws IOException, InterruptedException
+    {
+      // NOOP
+    }
+
+    @Override
+    public OutputCommitter getOutputCommitter(final TaskAttemptContext context)
+        throws IOException, InterruptedException
+    {
+      final MetadataStorageUpdaterJobHandler handler = getInjector(context.getConfiguration()).getInstance(
+          MetadataStorageUpdaterJobHandler.class
+      );
+
+      return new OutputCommitter()
+      {
+        @Override
+        public void setupJob(JobContext jobContext) throws IOException
+        {
+
+        }
+
+        @Override
+        public void setupTask(TaskAttemptContext taskContext) throws IOException
+        {
+
+        }
+
+        @Override
+        public boolean needsTaskCommit(TaskAttemptContext taskContext) throws IOException
+        {
+          return taskContext.getConfiguration().get(PUBLISHED_SEGMENT_KEY) != null;
+        }
+
+        @Override
+        public void commitTask(final TaskAttemptContext taskContext) throws IOException
+        {
+          final Progressable commitProgressable = new Progressable()
+          {
+            @Override
+            public void progress()
+            {
+              taskContext.progress();
+            }
+          };
+          final HadoopDruidConverterConfig config = converterConfigFromConfiguration(taskContext.getConfiguration());
+          final String finalSegmentString = taskContext.getConfiguration().get(PUBLISHED_SEGMENT_KEY);
+          if (finalSegmentString == null) {
+            throw new IOException("Could not read final segment");
+          }
+          final DataSegment newSegment = HadoopDruidConverterConfig.jsonMapper.readValue(
+              finalSegmentString,
+              DataSegment.class
+          );
+          log.info("Committing new segment");
+          taskContext.progress();
+          // Usually the indexing service handles its own commits, but standalone hadoop does not
+          if (shouldCommitMetadata(taskContext.getConfiguration())) {
```

Comment:
This is part of the task commit. At this point things are guaranteed to have been persisted to deep storage correctly, and the segment can be serviced independently. If it is desired to have the entire job be atomic, and not allow a completed segment to be serviced when others may not have been, then that is another concern.

Is there a reason why the job needs to be all or nothing?